crates/bevy_ecs/src/storage/blob_vec.rs

```diff
@@ -209,6 +209,29 @@ impl BlobVec {
         OwningPtr::new(self.swap_scratch)
     }
 
+    /// Removes the value at `index` and copies the value stored into `ptr`.
+    /// Does not do any bounds checking on `index`.
+    ///
+    /// # Safety
+    /// It is the caller's responsibility to ensure that `index` is < `self.len()`
+    /// and that `self[index]` has been properly initialized.
+    #[inline]
+    pub unsafe fn remove_unchecked(&mut self, index: usize, ptr: PtrMut<'_>) {
+        debug_assert!(index < self.len());
+        let last = self.len - 1;
```

Comment:
This would be clearer if last was defined right beside its usage; it took me a moment to figure out where this is used.

Reply:
Rewrote this to extract both the last and the target together. Might be a bit harder to follow, but the comments leave ample context now.